src/transformStateWithClones.js

```
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformedStates = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        stateCopy = addProperties(stateCopy, action);
+        transformedStates.push(stateCopy);
+        break;
+
+      case 'removeProperties' :
+        stateCopy = removeProperties(stateCopy, action);
+        transformedStates.push(stateCopy);
+        break;
+
+      case 'clear' :
+        stateCopy = {};
+        transformedStates.push(stateCopy);
+        break;
+
+      default :
+        return 'error';
```

let's throw an error, not just return a string